src/servers/src/postgres/handler.rs

```diff
@@ -177,10 +245,105 @@ fn type_translate(origin: &ConcreteDataType) -> Result<Type> {
     }
 }
 
+fn type_pg_to_gt(origin: &Type) -> Result<ConcreteDataType> {
+    // Note that we only support a small amount of pg data types
+    match origin {
+        &Type::BOOL => Ok(ConcreteDataType::boolean_datatype()),
+        &Type::CHAR => Ok(ConcreteDataType::int8_datatype()),
+        &Type::INT2 => Ok(ConcreteDataType::int16_datatype()),
+        &Type::INT4 => Ok(ConcreteDataType::int32_datatype()),
+        &Type::INT8 => Ok(ConcreteDataType::int64_datatype()),
+        &Type::VARCHAR | &Type::TEXT => Ok(ConcreteDataType::string_datatype()),
+        &Type::TIMESTAMP => Ok(ConcreteDataType::timestamp_datatype(
+            common_time::timestamp::TimeUnit::Millisecond,
+        )),
+        &Type::DATE => Ok(ConcreteDataType::date_datatype()),
+        &Type::TIME => Ok(ConcreteDataType::datetime_datatype()),
+        _ => error::InternalSnafu {
+            err_msg: format!("unimplemented datatype {origin:?}"),
+        }
+        .fail(),
+    }
+}
+
+#[derive(Default)]
+pub struct POCQueryParser;
+
+impl QueryParser for POCQueryParser {
+    type Statement = (Statement, String);
+
+    fn parse_sql(&self, sql: &str, types: &[Type]) -> PgWireResult<Self::Statement> {
+        let mut stmts = ParserContext::create_with_dialect(sql, &GenericDialect {})
```

Comment:
Is there a pg dialect we can use here?

Reply:
We can change this later. I'm not sure if using pg's dialect will introduce some changes in its parsed statement.  So here I reuse original dialect to keep the changeset minimal.